### ISSUE TITLE
automatic stream switching

### ISSUE THREAD

**Issue by ahmadabdolsaheb** [freeCodeCamp/coderadio-client#38]
With the new azuracast updates, relays have bitrate/format, unique identifiers, and per-stream listener counts.

the main strategy would be to directing listeners towards the relays.

here is the suggested logic

if relays exist choose a fast relays with least listeners.
if relays exist and speed is low, choose a low bitrate relay with least listeners.

if no relays choose fast non-relay
if relays exist and speed is low, choose low bitrate non-relay.

error handling on loading:
- If > 220, use the other streams
- all relay non-relay high bitrate streams 403ing? Drop the new user to 64
- On 403 during playback? Use another stream.

here are some of the issues I experienced this week:

GET https://coderadio-relay.freecodecamp.org/radio/8010/radio.mp3 net::ERR_INCOMPLETE_CHUNKED_ENCODING 200 (OK)

Failed to load resource: net::ERR_INCOMPLETE_CHUNKED_ENCODING

**Comment by huyenltnguyen** [comment]
I'll grab this one

**Comment by ahmadabdolsaheb** [comment]
@huyenltnguyen how is your progress on this issue? please let u know if you need any clarification or assistance to arrive at a solution.

**Comment by huyenltnguyen** [comment]
@ahmadabdolsaheb Oh, sorry. Life got in the way before I had a chance to look into this. Is the task urgent? I think I can work on it this weekend.

So sorry, I should have kept you posted.

**Comment by ahmadabdolsaheb** [comment]
It is not urgent, but it could really improve the user experience.

**Comment by huyenltnguyen** [comment]
@ahmadabdolsaheb I lied. I wasn't able to work on this last week 😕 

Anyway, I raised a draft PR for the issue. In the PR, I've updated the stream sorting logic to include number of listeners as one of the sort criteria. However, I'm going to need some help with the error handling. My questions are:
- How to reproduce the errors?
- Where is the right place in the code to catch an error? I'm guessing that it should be in `play()` (https://github.com/freeCodeCamp/coderadio-client/blob/master/src/components/App.js#L109)?
- Is it correct that we will use `SUB.on('error')` for error handling? (I got that from https://github.com/slact/nchan.js) 

Thanks in advance, and sorry again for the delay.
